astropy/coordinates/angles/core.py

```diff
@@ -404,7 +404,7 @@ def _wrap_at(self, wrap_angle):
         # than Quantity objects for speed.
         a360 = u.degree.to(self.unit, 360.0)
         wrap_angle = wrap_angle.to_value(self.unit)
-        self_angle = self.view(np.ndarray)
+        self_angle = self.view(np.ndarray).astype(f"f{self.dtype.itemsize}", copy=False)
```

Comment:
originally I just cast to `float64` but then got failures in tests exercising `float32`, so this more complex casting ensures that preserve floating point precision, while still enabling int->float conversion

Reply:
This doesn't work - it has to be a view in order to do the in-place wrapping. I think we'll need to forbid integer.

Reply:
good point. Let me wrap up an alternative PR then and see if it breaks more than one test

Reply:
see #16220